src/pseudo/PspUpf.jl

```diff
@@ -0,0 +1,251 @@
+using LinearAlgebra
+using Interpolations: linear_interpolation
+using PseudoPotentialIO: load_upf
+
+struct PspUpf{T} <: NormConservingPsp
+    Zion::Int                             # Ionic charge (Z - valence electrons)
+    lmax::Int                             # Maximal angular momentum in the non-local part
+    
+    rgrid::Vector{T}                      # Radial grid
+    drgrid::Vector{T}                     # Radial grid derivative / integration factor
+    
+    vloc::Vector{T}                       # Local potential on the radial grid
+    vloc_interp                           # Local potential interpolator
```

Comment:
Why is this not typed as well?

Reply:
I just didn't understand how best to type it when this is an example of one of the types:

```
Interpolations.Extrapolation{Float64, 1, Interpolations.GriddedInterpolation{Float64, 1, Vector{Float64}, Interpolations.Gridded{Interpolations.Linear{Interpolations.Throw{Interpolations.OnGrid}}}, Tuple{Vector{Float64}}}, Interpolations.Gridded{Interpolations.Linear{Interpolations.Throw{Interpolations.OnGrid}}}, Interpolations.Throw{Nothing}}
```